sw/device/sca/ecc256_keygen_serial.c

```diff
@@ -55,8 +56,44 @@ enum {
    * Mode option for the ECDSA keygen app (generates the full keypair).
    */
   kEcc256ModeKeypair = 2,
+  /**
+   * Max number of traces per batch.
+   */
+  kNumBatchOpsMax = 256,
 };
 
+/**
+ * An array of seeds to be used in a batch
+ */
+uint32_t batch_seeds[kNumBatchOpsMax][kEcc256SeedNumWords];
+
+/**
+ * An array of masks to be used in a batch
+ */
+uint32_t batch_masks[kNumBatchOpsMax][kEcc256SeedNumWords];
```

Comment:
I assume that it would work to allocate the required space locally. The required space depends on how many samples for each trace are configured, but is at the end limited through the husky's buffer size.
But similar to the max number of traces per batch, I just looked at the kmac and sha3 batch modes and there it is implemented the same way. (Which is no reason to not change it, but then it probably might be a good Idea to change it in the other batch modes as well.)

Reply:
If I recall correctly, in the earlier versions of kmac capture setup, `batch_keys` and `batch_messages` were set inside the `for` loop. This was resulting in a strange problem with "fake leakage" due to residual charge on one of the on-board capacitors.
It's probably still possible to get rid of the global arrays but we have to be careful to do it in an SCA-friendly way.
I would leave this as it is for now and continue discussion in #17441